pycbc/inference/io/posterior.py

```diff
@@ -33,48 +33,55 @@ class PosteriorFile(BaseInferenceFile):
     name = 'posterior_file'
 
     def read_raw_samples(self, fields, **kwargs):
-        samples = self[self.samples_group]
-        return {field: samples[field][:] for field in fields}
+        return read_raw_samples_from_file(self, fields, **kwargs)
 
     def write_samples(self, samples, parameters=None):
-        """Writes samples to the given file.
+        return write_samples_to_file(self, samples, parameters=None)
```

Comment:
You need to pass `parameters=parameters` here. As is, the parameters input would be ignored.